src/helics/shared_api_library/helics.h

```diff
@@ -632,7 +632,7 @@ HELICS_EXPORT void helicsFederateInfoSetBrokerInitString(helics_federate_info fi
  * @param[in,out] err An error object that will contain an error code and string if any error occurred during the execution of the function.
  * @endforcpponly
  */
-HELICS_EXPORT void helicsFederateInfoSetCoreType(helics_federate_info fi, int coretype, helics_error* err);
+HELICS_EXPORT void helicsFederateInfoSetCoreType(helics_federate_info fi, helics_core_type coretype, helics_error* err);
```

Comment:
in HELICS 3 (at some point) we will allow user defined core types in which case helics can't know about all possibilities

Reply:
Is `helics_core_type` something that could become a typedef, with a separate enum (`helics_builtin_core_types`) for any that come with HELICS?